packages/styles/scss/components/number-input/_number-input.scss

```
@@ -499,17 +507,31 @@
 
     padding-inline-end: convert.to-rem(145px);
   }
+
   // Skeleton State
+
   .#{$prefix}--number.#{$prefix}--skeleton {
     @include skeleton;
 
-    block-size: 2.5rem;
     inline-size: 100%;
 
     input[type='number'] {
       display: none;
     }
   }
+
+  .#{$prefix}--number.#{$prefix}--skeleton.#{$prefix}--number--sm {
+    block-size: convert.to-rem(32px);
+  }
+
+  .#{$prefix}--number.#{$prefix}--skeleton.#{$prefix}--number--md {
+    block-size: convert.to-rem(40px);
+  }
+
+  .#{$prefix}--number.#{$prefix}--skeleton.#{$prefix}--number--lg {
+    block-size: convert.to-rem(48px);
```

```suggestion
    block-size: $spacing-07;
  }

  .#{$prefix}--number.#{$prefix}--skeleton.#{$prefix}--number--md {
    block-size: $spacing-08;
  }

  .#{$prefix}--number.#{$prefix}--skeleton.#{$prefix}--number--lg {
    block-size: $spacing-09;
```

Can we use tokens instead?